### PR TITLE
Experiment section intro

### PR DESCRIPTION
**Pull request by Pattrigue**
Some starting paragraphs on the experiments. Will need to add some more and possibly rewrite a bit once experiments have been performed.

### CODE REVIEW

File: report_pre_thesis/src/sections/methodology.tex [hunk 1]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+To evaluate the performance of each of the components in the pipeline, we focus our experiments on three main aspects:
+
+\begin{itemize}
+	\item \textbf{Outlier removal} to assess the impact of leaving outliers in the dataset or using a different outlier removal method.
+	\item \textbf{Hyperparameter tuning} to assess the impact of different hyperparameter configurations.
+	\item \textbf{Other models} to compare the performance of the PLS1-SM and ICA models to other models.
+\end{itemize}
+
+\noindent
+We recognize that these three aspects are commonly used to evaluate the performance and identify potential improvements of a machine learning pipeline.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:223]: Commonly used to evaluate? Not really.

These are typically done during the explorative phase of a ML project, where you are trying to find the most problem model and dataset configuration. 
During such a stage, you'd evaluate many models on a validation set.
Few would likely pass your tests, at which point they would be considered for further optimization (e.g., extensive hyperparameter tuning, etc.).
Finally you'd get the end model's performance given some metric(s) - RMSE for us.

We need to be rather sharp when reasoning about our choices regarding experiments, given their importance to the project.

File: report_pre_thesis/src/sections/methodology.tex [hunk 2]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+However, since the original authors did not conduct any experiments with alternative methods to demonstrate that their selected approach was superior, we feel that we have an incomplete picture of the potential performance of the pipeline.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:224]: Need to be more precise about alternative methods. They did tune hyperparameters. Do we mean in terms of models, algorithms, architectures?

File: report_pre_thesis/src/sections/methodology.tex [hunk 3]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+Additionally, a benefit of experimenting with alternative methods is that we can uncover which components contribute the most to the overall error and therefore would benefit the most from further research and development.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:225]: This cannot be an "additional" point. It is the central, most important reason for our experiments.

File: report_pre_thesis/src/sections/methodology.tex [hunk 4]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+
+% TODO: mention the reason we are doing experiments (to try and align our replication with the original pipeline as closely as possible)
+
+\subsubsection{Experiment: Outlier Removal}\label{sec:experiment_outlier_removal}
+The original PLS1-SM identified outliers manually by inspecting the leverage and spectral residuals plots.
+However, we have chosen to automate this based on the reasons described in Section~\ref{sec:methodology_outlier_removal}.
+It would therefore be interesting to see how the performance of the pipeline is affected by adjusting this process.
+Firstly, to see if a more or less conservative approach would be beneficial, it would be interesting to try out different significance levels for the chi-squared test.
+Secondly, it would be interesting to see how the performance is affected by omitting outlier removal altogether.
+This is a reasonable experiment because of the extensive efforts invested in developing the ChemCam calibration dataset, suggesting a low presence of significant outliers.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:235]: Can we lead with the reason? Saying that it's interesting is weaker than leading with the reason we provide right after. And perhaps either ref back to where we first say this, or mention that this was a point from a CCAM team member.

File: report_pre_thesis/src/sections/methodology.tex [hunk 5]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+We recognize that these three aspects are commonly used to evaluate the performance and identify potential improvements of a machine learning pipeline.
+However, since the original authors did not conduct any experiments with alternative methods to demonstrate that their selected approach was superior, we feel that we have an incomplete picture of the potential performance of the pipeline.
+Additionally, a benefit of experimenting with alternative methods is that we can uncover which components contribute the most to the overall error and therefore would benefit the most from further research and development.
+
+% TODO: mention the reason we are doing experiments (to try and align our replication with the original pipeline as closely as possible)
+
+\subsubsection{Experiment: Outlier Removal}\label{sec:experiment_outlier_removal}
+The original PLS1-SM identified outliers manually by inspecting the leverage and spectral residuals plots.
+However, we have chosen to automate this based on the reasons described in Section~\ref{sec:methodology_outlier_removal}.
+It would therefore be interesting to see how the performance of the pipeline is affected by adjusting this process.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:232]: You say interesting three times in three sentences. Consider more precise language?

File: report_pre_thesis/src/sections/methodology.tex [hunk 6]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+Firstly, to see if a more or less conservative approach would be beneficial, it would be interesting to try out different significance levels for the chi-squared test.
+Secondly, it would be interesting to see how the performance is affected by omitting outlier removal altogether.
+This is a reasonable experiment because of the extensive efforts invested in developing the ChemCam calibration dataset, suggesting a low presence of significant outliers.
+
+For the ICA phase, the original authors used Median Absolute Deviation for outlier removal, but the specifics of their approach remain unclear.
+Therefore, we omitted outlier removal in the ICA phase of our pipeline.
+However, implementing outlier removal using MAD would still be an interesting experiment to see if it improves the performance of the pipeline.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:239]: I think we need some other way to say interesting in the experiments section. Some other way to motivate picking experiments. This doesn't reason about why, it just says we'd think it's interesting.

File: report_pre_thesis/src/sections/methodology.tex [hunk 7]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+To this extent, we would try to align the implementation with the original authors' approach as closely as possible, and also experiment with different significance levels for the MAD test.
+
+% TODO: If there's time, include other outlier removal methods as well, such as Isolation Forest, Local Outlier Factor, etc.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:242]: No. There isn't time. Other experiments would take precedence.

File: report_pre_thesis/src/sections/methodology.tex [hunk 8]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+
+\subsubsection{Experiment: Hyperparameter Tuning}\label{sec:experiment_hyperparameter_tuning}
+\citet{cleggRecalibrationMarsScience2017} use qualitative judgement to identify hyperparameters for their PLS1-SM model.
+This can be an error-prone approach, as it is difficult to ensure that the selected hyperparameters are optimal.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:246]: Cite?

File: report_pre_thesis/src/sections/methodology.tex [hunk 9]
```diff
@@ -206,4 +208,63 @@ \subsection{MOC}\label{sec:methodology_moc}
...
+For the PLS1-SM model we decided to use the common grid search algorithm for testing different hyperparameters for the PLS models.
+% Explain set up...
+
+Since each independent component does not necessarily correlate one-to-one with the number of elements that one wishes to identify in a spectra, we decided to experiment with a number of components ranging between 4 and 25.
+This range is within the vicinity of the original selection of components whilst providing us with a set of reasonable extremes.
+
+% Probably show the setup in some way
+
+\subsubsection{Experiment: Other Models}\label{sec:experiment_other_models}
+\citet{cleggRecalibrationMarsScience2017} have not demonstrated any experiments with alternative methods to demonstrate that their selected approach is superior.
```
Comment (chhoumann) [inline report_pre_thesis/src/sections/methodology.tex:261]: Besides their original PLS method used from pre-flight to a few years after.

**Review by chhoumann — CHANGES_REQUESTED** [review]

File: report_pre_thesis/src/sections/methodology.tex (hunk 7 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Removed the comment.

File: report_pre_thesis/src/sections/methodology.tex (hunk 3 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Should now be rephrased to address this point.

File: report_pre_thesis/src/sections/methodology.tex (hunk 5 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Rephrased.

File: report_pre_thesis/src/sections/methodology.tex (hunk 4 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Changed the order and referenced back to when we previously described the efforts involved in developing the CCAM calib data.

File: report_pre_thesis/src/sections/methodology.tex (hunk 6 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Rephrased this.

File: report_pre_thesis/src/sections/methodology.tex (hunk 1 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Rephrased it, should be addressed now.

File: report_pre_thesis/src/sections/methodology.tex (hunk 2 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Should be addressed now.

File: report_pre_thesis/src/sections/methodology.tex (hunk 8 above)
Comment (Pattrigue) [reply report_pre_thesis/src/sections/methodology.tex]: Rephrased it.

**Review by chhoumann — APPROVED** [review]

**Review by chhoumann — APPROVED** [review]
